lib/transformer.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const aws = require('aws-sdk')
+const zlib = require('zlib')
+
+const formatter = require('./formatter.js')
+
+const BATCH_LIMIT = parseInt(process.env.LOGDNA_BATCH_LIMIT) || 25
+const s3 = new aws.S3({
+  apiVersion: '2006-03-01'
+})
+
+const batchify = (logs) => {
+  const batches = []
+  let batch = [], batch_size = 0
+  logs.forEach((log) => {
+    batch.push(log)
+    batch_size += 1
+    if (batch_size >= BATCH_LIMIT) {
+      batches.push(batch)
+      batch = []
+      batch_size = 0
+    }
+  })
+
+  if (batch_size > 0) { batches.push(batch) }
+  return batches
+}
```

Comment:
https://github.com/answerbook/stdlib-node/blob/master/lib/array/chunk.js

```js
  const size = BATCH_LIMIT
  const out = []
  for (let i = 0; i < arr.length; i += size) {
    out.push(arr.slice(i, i + size))
  }
  return out
```

we should avoid iterating with `forEach`, there's no need for an additional function call when a `for` or `for..of` will do the same, unless you really need the closure (which we do not here)

Reply:
Done in `utils`